go-controller/pkg/ovn/base_network_controller_pods.go

```diff
@@ -561,15 +561,25 @@ func (bnc *BaseNetworkController) addLogicalPortToNetwork(pod *kapi.Pod, nadName
 		return nil, nil, nil, false, err
 	}
 
-	// set addresses on the port
-	// LSP addresses in OVN are a single space-separated value
+	lsp.Enabled = enable
+	if lsp.Enabled != nil {
+		customFields = append(customFields, libovsdbops.LogicalSwitchPortEnabled)
+	}
+
 	addresses = []string{podAnnotation.MAC.String()}
 	for _, podIfAddr := range podAnnotation.IPs {
 		addresses[0] = addresses[0] + " " + podIfAddr.IP.String()
 	}
 
-	lsp.Addresses = addresses
-	customFields = append(customFields, libovsdbops.LogicalSwitchPortAddresses)
+	// Skip address configuration if LSP is disabled since it will install
+	// l2 look up flows that harms some topologies
```

Comment:
Can you provide more context around why do you need to remove the address and how is it specific to localnet?

Reply:
At localnet there is no remote LSP referencing the remote virtual machine port so moving traffic from local to remote is done over the underlay, we found that moving to Enable=false does not clean up the OVN l2 lookup table (more or less the switch mac learning) so it still tries to send traffic to the old LSP since the table is still pointing to it, the fix is to remove the LSP Address so the mac is removed and OVN remove the entry at the l2 lookup table.

Reply:
Yes, this is more of an OVN thing.  I'm not sure if it will be fixed in OVN but removing the mac and IP for disabled ports should be OK.

Reply:
In fact we were thinking after fixing that, that maybe that's what we should do instead of playing with Enabled field, but we don't want to change that now.